### PR TITLE
Add events/getUsers

### LINKED ISSUE #31: events/getUsers
A way to get user data for users registered for an event

### PR DESCRIPTION
**Pull request by ianmah**
A function that returns an array of user data for users going to the event specified

Had to update node version to support `.flatMap()`

Closes #31

### CODE REVIEW

File: handlers/helpers.js [hunk 1]
```diff
@@ -1,83 +1,110 @@
-const AWS = require('aws-sdk');
```
Comment (ianmah) [inline handlers/helpers.js]: Ran this file through prettier, but added a new function `batchGet`

File: handlers/event.js [hunk 2]
```diff
@@ -88,6 +87,73 @@ module.exports.get = async (event, ctx, callback) => {
 
 };
 
+module.exports.getUsers = async (event, ctx, callback) => {
+
+  const id = event.queryStringParameters.id;
```
Comment (joshflchan) [inline handlers/event.js]: change to eventID to avoid confusion with user id?

File: handlers/event.js [hunk 3]
```diff
@@ -88,6 +87,73 @@ module.exports.get = async (event, ctx, callback) => {
...
+     */
+    let keysForRequest = registrationList.map(registrationObj => {
+      let keyEntry = {}
+      keyEntry.id = parseInt(registrationObj.id)
+      return keyEntry
+    })
+    console.log('Keys:', keysForRequest)
+
+    let keyBatches = [];
+    const size = 100 // max BatchGetItem count
```
Comment (joshflchan) [inline handlers/event.js]: the maximum number of users we can get is 100?

File: handlers/event.js [hunk 4]
```diff
@@ -88,6 +87,73 @@ module.exports.get = async (event, ctx, callback) => {
...
+    while (keysForRequest.length > 0) {
+      keyBatches.push(keysForRequest.splice(0, size))
+    }
+
+    await Promise.all(keyBatches.map(batch => {
+      return helpers.batchGet(batch, 'biztechUsers' + process.env.ENVIRONMENT)
+    })).then(result => {
+      const results = result.flatMap(batchResult => batchResult.Responses.biztechUsers)
```
Comment (joshflchan) [inline handlers/event.js]: So `flatMap` is being used to get the data of each user in the list of batches

File: handlers/event.js (hunk 3 above)
Comment (ianmah) [reply handlers/event.js]: Per call to `docClient.batchGet`, yes. So we have to group ids into batches of 100

File: handlers/helpers.js [hunk 5]
```diff
@@ -1,83 +1,110 @@
...
-                    updateExpression += key + '\= :' + key + ',';
-                    expressionAttributeValues[':' + key] = obj[key];
-                }
-            }
+  /**
+   *
+   * @param {Array} batch - List of batches in form of [{Key: value}]
+   * @param {String} tableName - Name of table to call batchGet
+   */
+  batchGet: function(batch, tableName) {
```
[the remark this answers is not in the document]
Comment (ianmah) [reply handlers/helpers.js]: `batchGet` requests take data in the form of `keyName: keyValue` so in the case of getUsers a batch is a list of `id: [id]`

[the remark this answers is not in the document]
Comment (ianmah) [reply handlers/helpers.js]: Aka if we had a table with primary key `studentID` and wanted to find the student with id `777` a batch would be `[{studentID: 777}]`

File: handlers/event.js (hunk 4 above)
Comment (ianmah) [reply handlers/event.js]: We have an array of batch calls, which returns an array of promises, when evaluated we have an array of arrays of batch get response data. Flatmap combines that into an array of user data

File: handlers/event.js (hunk 2 above)
Comment (ianmah) [reply handlers/event.js]: All of the other `events/` endpoints use `id`. Imo we keep them the same? Do you think we should change all of them to use `eventID`?

File: handlers/helpers.js [hunk 6]
```diff
@@ -1,83 +1,110 @@
...
+    return docClient.batchGet(batchRequestParams).promise()
+  },
+
+  /**
+   *
+   * @param {*} id - String or Integer item ID
+   * @param {Object} obj - object containing key value paris
+   * @param {String} table - name of table, ie 'biztechUsers'
+   */
+  updateDB: async function(id, obj, table) {
```
Comment (jacqueschen1) [inline handlers/helpers.js]: Did you change updateDB or is this just git being funny?

Comment (ianmah) [reply handlers/helpers.js]: I just ran prettify on this file

File: handlers/event.js (hunk 2 above)
Comment (ianmah) [reply handlers/event.js]: Yeah same, I like that

**Pull request merged**